cli/cli-client/src/main/java/org/infinispan/cli/commands/User.java

```diff
@@ -0,0 +1,271 @@
+               String confirm = null;
+               while (confirm == null || !confirm.equals(password)) {
+                  confirm = invocation.getShell().readLine(new Prompt(MSG.userToolPasswordConfirm(), '*'));
+               }
+            } catch (InterruptedException e) {
+               return CommandResult.FAILURE;
+            }
+         }
+         userTool.createUser(username, password, realm, plainText, groups, algorithms);
+         return CommandResult.SUCCESS;
+      }
+   }
+
+   @CommandDefinition(name = Describe.CMD, description = "Describes a user")
+   public static class Describe extends CliCommand {
+      public static final String CMD = "describe";
+
+      @Argument(description = "The username for the user", required = true)
+      String username;
+
+      @Option(description = "The path of the users.properties file", name = "users-file", shortName = 'f')
+      String usersFile;
+
+      @Option(description = "The path of the groups.properties file", name = "groups-file", shortName = 'w')
+      String groupsFile;
+
+      @Option(description = "The server root", defaultValue = "server", name = "server-root", shortName= 's')
+      String serverRoot;
+
+      @Option(shortName = 'h', hasValue = false)
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      public CommandResult exec(ContextAwareCommandInvocation invocation) throws CommandException {
+         UserTool userTool = new UserTool(serverRoot, usersFile, groupsFile);
+         invocation.getShell().writeln(userTool.describeUser(username));
+         return CommandResult.SUCCESS;
+      }
+   }
+
+   @CommandDefinition(name = Remove.CMD, description = "Removes a user", aliases = "rm")
+   public static class Remove extends CliCommand {
+      public static final String CMD = "remove";
+
+      @Argument(description = "The username for the user", required = true)
+      String username;
+
+      @Option(description = "The path of the users.properties file", name = "users-file", shortName = 'f')
+      String usersFile;
+
+      @Option(description = "The path of the groups.properties file", name = "groups-file", shortName = 'w')
+      String groupsFile;
+
+      @Option(description = "The server root", defaultValue = "server", name = "server-root", shortName= 's')
+      String serverRoot;
+
+      @Option(shortName = 'h', hasValue = false)
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      public CommandResult exec(ContextAwareCommandInvocation invocation) throws CommandException {
+         UserTool userTool = new UserTool(serverRoot, usersFile, groupsFile);
+         userTool.removeUser(username);
+         return CommandResult.SUCCESS;
+      }
+   }
+
+   @CommandDefinition(name = Modify.CMD, description = "Modifies a user")
+   public static class Modify extends CliCommand {
```

Comment:
wdyt of splitting this subcommand in 2? 
* one for change password (set new password, encrypt with new algorithm)
* another one for groups (add or remove groups)

Reply:
Good idea: I've done that.
`user password` will change a user's password
`user groups` will change the groups